next step for the big change, product, cart and order need to be suit…

eilonasher opened the pull request:
…e for the last version we talked about:

account need to be: name, uid(unique value), cart, static currentAccount
productActivity file: changes that support in the change above and update all the code according to the change

Commit:
next step for the big change, product, cart and order need to be suite for the last version we talked about:
account need to be: name, uid(unique value), cart, static currentAccount
productActivity file: changes that support in the change above and update all the code according to the change

galpinto25 commented on app/src/main/java/com/example/sabonit/Account.java:
```diff
@@ -46,7 +46,7 @@ public Account(String name, String uid)
 
     public static Account getCurrentAccount() {
-        return currentAccount;
+        return Account.currentAccount;
```
I think that these two lines are doing the same

eilonasher replied:
youre right, i changed it because i thought this is a problem but it doesnt matter whats written here

yairelboher commented on app/src/main/java/com/example/sabonit/Account.java:
```diff
@@ -46,7 +46,7 @@ public Account(String name, String uid)
 
     public static Account getCurrentAccount() {
-        return currentAccount;
+        return Account.currentAccount;
```
try to revert this code line to the red colored version, it seems to have equal outcome

eilonasher replied:
ok ill change it in the next version, see my answer to gal

galpinto25 commented on app/src/main/java/com/example/sabonit/CategoriesActivity.java:
```diff
@@ -37,16 +37,19 @@ public void intentDepartment(View view) {
+                intent.putExtra("Department", "Face & Body Wash");
                 break;
             case R.id.hand_soap:
-                intent.putExtra("Department", "Hand Soap ");
+                intent.putExtra("Department", "HandSoap");
                 break;
             case R.id.house_cleaning:
-                intent.putExtra("Department", "House Cleaning ");
+                intent.putExtra("Department", "House Cleaning");
                 break;
             case R.id.laundry:
-                intent.putExtra("Department", "Laundry ");
+                intent.putExtra("Department", "Laundry");
+                break;
+            case R.id.dishwashing_liquid:
+                intent.putExtra("Department", "Dishwashing Liquid");
```
OK, great now we are doing the intent with meaningful department name

galpinto25 commented on app/src/main/java/com/example/sabonit/ProductActivity.java:
```diff
@@ -28,15 +28,11 @@
 public class ProductActivity extends AppCompatActivity {
 
-    private final int NEW_BOTTLE = -1;
+    private final int NEW_BOTTLE = 10;
```
Great, for now the maximum will be 10
We'll change it in the future

galpinto25 commented on app/src/main/java/com/example/sabonit/ProductActivity.java:
```diff
@@ -138,31 +134,50 @@ private void fillBottle(double curLitersInSeekBar, int color) {
+            if (currentOrder.getProduct().getDepartment().equals(department))
             {
-                initalProgress = (int) (currentOrder.getLiters() * 10.0);
+                currentProduct = currentOrder.getProduct();
```
Great, now the product is being updated, if it was ordered before

galpinto25 commented on app/src/main/java/com/example/sabonit/ProductActivity.java:
```diff
@@ -138,31 +134,50 @@ private void fillBottle(double curLitersInSeekBar, int color) {
+        updateCheckBoxNewBottle();
+    }
 
+    private void updateSmellRadioButton(String smell) {
```
Great, now the radio buttons will be updated as well

galpinto25 commented on app/src/main/java/com/example/sabonit/ProductActivity.java:
```diff
@@ -246,21 +263,26 @@ public void onCheckboxClicked(View view)
             litersTitle.setVisibility(View.VISIBLE);
             currentProductLiters = oldLitters;
         }
+        fillBottle(currentProductLiters, bottleColor);
+        litersTitle.setText(String.valueOf(currentProductLiters) + "L");
+        literSeekBar.setProgress((int) (currentProductLiters * 10));
```
Maybe can be divided to a new function

eilonasher replied:
i'll do it

yairelboher commented on app/src/main/java/com/example/sabonit/ProductActivity.java:
```diff
@@ -28,15 +28,11 @@
 public class ProductActivity extends AppCompatActivity {
 
-    private final int NEW_BOTTLE = -1;
+    private final int NEW_BOTTLE = 10;
```
this value is fit to the max value of the seek bar, but maybe we will define different max values for the different departments.
im not sure, but in my opion the solution for that is to use the function getMaximum(), that returns the progress bar's maximum value

eilonasher replied:
ok ill change it

yairelboher commented on app/src/main/java/com/example/sabonit/ProductActivity.java:
```diff
@@ -70,20 +65,20 @@ protected void onCreate(Bundle savedInstanceState) {
         {
             public void onCheckedChanged(RadioGroup group, int checkedId) {
-                changeScent();
+                changeSmell();
```
you have the permission to decide the appropriate name, although Im not sure it's the best decision.
just verify that we all use the name "smell" instead of "scent"

eilonasher replied:
i'll check it, the decision is to be smell everywhere in the code

yairelboher commented on app/src/main/java/com/example/sabonit/ProductActivity.java:
```diff
@@ -46,10 +42,9 @@
-    private int initalProgress;
+    private int initialProgress;
     private SeekBar literSeekBar;
     private RadioGroup radioScentGroup;
```
needs to be changed to smell

eilonasher replied:
ok

Commit:
changes according to your code review:
account - change the line of account. to be without it
scent to smell
constant newBottle deleted
fix bug of reading isProductNewBottle

galpinto25 reviewed (commented):
Great work!
